bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/installcomponents/container/ComponentContainer.java

```diff
@@ -0,0 +1,153 @@
+		controlGroup.setLayoutData(layoutData);
+		controlGroup.setBackground(new Color(240, 233, 233));
+		controlGroup.setText(componentVO.getName().toUpperCase());
+		Font boldFont = new Font(controlGroup.getDisplay(), new FontData("Arial", 8, SWT.BOLD)); //$NON-NLS-1$
+		controlGroup.setFont(boldFont);
+
+		if (componentVO.getComponentDetails() != null && componentVO.getComponentDetails().getDescription() != null)
+		{
+			detailsText = new Text(controlGroup, SWT.MULTI | SWT.WRAP);
+			detailsText.setLayoutData(new GridData(GridData.FILL_BOTH));
+			detailsText.setBackground(new Color(240, 233, 233));
+			detailsText.setText(componentVO.getComponentDetails().getDescription());
+		}
+
+		if (componentVO.getComponentDetails() != null && componentVO.getComponentDetails().getTargets() != null)
+		{
+			targetsLabel = new Label(controlGroup, SWT.NONE);
+			StringBuilder sbTargets = new StringBuilder();
+			targetsLabel.setBackground(whiteColor);
+			sbTargets.append(componentVO.getComponentDetails().getTargets().get(0));
+
+			for (int i = 0; i < componentVO.getComponentDetails().getTargets().size(); i++)
+			{
+				sbTargets.append(", "); //$NON-NLS-1$
+				sbTargets.append(componentVO.getComponentDetails().getTargets().get(i));
+			}
+
+			targetsLabel.setText(sbTargets.toString());
+		}
+
+		if (componentVO.getComponentDetails() != null
+				&& !StringUtil.isEmpty(componentVO.getComponentDetails().getVersion()))
+		{
+			versionLabel = new Label(controlGroup, SWT.NONE);
+			versionLabel.setBackground(new Color(240, 233, 233));
+			versionLabel.setText(componentVO.getComponentDetails().getVersion());
+		}
+
+		Composite btnComposite = new Composite(controlGroup, SWT.NONE);
+		btnComposite.setBackground(new Color(240, 233, 233));
+		btnComposite.setLayout(new GridLayout(2, true));
+
+		openReadMe = new Button(btnComposite, SWT.PUSH);
+		openReadMe.setText(Messages.InstallComponents_OpenReadmeButton);
+		openReadMe.setBackground(whiteColor);
+		openReadMe.addSelectionListener(new SelectionAdapter()
+		{
+			@Override
+			public void widgetSelected(SelectionEvent e)
+			{
+				String url = componentVO.getComponentDetails().getReadMe();
+				try
+				{
+					Desktop.getDesktop().browse(new URL(url).toURI());
+				}
+				catch (Exception e1)
+				{
+					Logger.log(e1);
+				}
+			}
+		});
+
+		installButton = new Button(btnComposite, SWT.PUSH);
+		installButton.setText(Messages.InstallComponents_InstallButton);
+		if (componentVO.isComponentAdded())
+		{
+			installButton.setText(Messages.InstallComponents_InstallButtonAlreadyAdded);	
+			installButton.setEnabled(false);
+		}
+		installButton.setBackground(whiteColor);
+		InstallCommandHandler installCommandHandler = new InstallCommandHandler(componentVO.getName(),
+				componentVO.getNamespace(),
+				componentVO.getComponentDetails() != null && componentVO.getComponentDetails().getVersion() != null
+						? componentVO.getComponentDetails().getVersion()
```

Comment:
We could improve code readability by assigning `componentVO.getComponentDetails()` to variable and using it in three places

Reply:
Done